### DISCUSSION TITLE
How to add to a Plotter two meshes to be drawn with the same origin.

### DISCUSSION THREAD

**Discussion by juanccilleruelo** [pyvista/pyvista#3960] [Q&A]
Hello again!

I'm adding to a plotter two mesh generated from a pv.Cube, but when there are shown, both appear centered. 
I need that each one appears located on the same X, Y, and Z coordinates. Not centered.

Thank you in advance!

**Comment by MatthewFlamm** [comment]
Do you mean that you want one of the corners to be coincident?  You need to adjust the `center` parameter or possibly more useful for your case is to use the `bounds` argument instead.

https://docs.pyvista.org/api/utilities/_autosummary/pyvista.Cube.html

**Reply by juanccilleruelo** [reply]
I want that all my meshes were referenced to a common point (0, 0, 0) in Euclidean X, Y, and Z coordinates.

Do you have an example with "center" or "bounds"? Please. I've read almost all the documentation available, but this doesn't help, at least in the first stage of the learning process. 

Thank's.

**Reply by MatthewFlamm** [reply]
Here is an example that places two cubes each with one vertex at the origin.

```py
import pyvista as pv
mesh1 = pv.Cube(center=(0.5, 0.5, 0.5))  # Side Length is 1 by default
mesh2 = pv.Cube(bounds=(0., 2.0, 0., 2.0, 0., 2.0))
pl = pv.Plotter()
pl.add_mesh(mesh1, opacity=0.5, color='red')
pl.add_mesh(mesh2, opacity=0.5, color='blue')
pl.add_axes_at_origin()
pl.show()
```

![image](https://user-images.githubusercontent.com/39341281/217315704-ac59910e-2d87-489d-9d25-5a5dc1179241.png)

**Reply by juanccilleruelo** [reply]
If I understand well, the conclusion is that I need to define the center of the piece at a point sited at the half distance of his measures in each axe. 

```py
import pyvista as pv
plotter = pv.Plotter(window_size = (1890, 900)) 

X = 6 
Y = 7 
Z = 8 
Cube1 = pv.Cube(center=(X/2, Y/2, Z/2), x_length = X, y_length = Y, z_length = Z) 

X = 3 
Y = 3 
Z = 3 
Cube2 = pv.Cube(center=(X/2, Y/2, Z/2), x_length = X, y_length = Y, z_length = Z) 

X = 12 
Y = 8 
Z = 15
Cube3 = pv.Cube(center=(X/2, Y/2, Z/2), x_length = X, y_length = Y, z_length = Z) 

plotter.add_mesh(Cube1, opacity = 0.5, color = 'red' , show_edges = True) 
plotter.add_mesh(Cube2, opacity = 0.5, color = 'blue', show_edges = True) 
plotter.add_mesh(Cube3, opacity = 0.5, color = 'blue', show_edges = True) 

plotter.show_bounds(grid = 'front', location = 'outer', all_edges = True) 

plotter.add_axes_at_origin(labels_off = True) 
               
plotter.show()       
```

![image](https://user-images.githubusercontent.com/10785381/217597790-81d51eb9-0bd9-471e-967b-4429d18126a4.png)



Thank you @MatthewFlamm

**Reply by MatthewFlamm** [reply]
> If I understand well, the conclusion is that I need to define the center of the piece at a point sited at the half distance of his measures in each axe.

If you want one Cube vertex to be at the origin, yes.  Or you can use `bounds` argument and place either the min or the max bounds at `(0, 0, 0)`.  I gave both examples in my code.

**Reply by juanccilleruelo** [reply]
> > If I understand well, the conclusion is that I need to define the center of the piece at a point sited at the half distance of his measures in each axe.
> 
> If you want one Cube vertex to be at the origin, yes. Or you can use `bounds` argument and place either the min or the max bounds at `(0, 0, 0)`. I gave both examples in my code.

In Euclidean space, x, y, and z coordinates are more accurate than the use of six points in the space (bounds). 

It is easier, at least for me, to think in (0, 0, 0) as the origin of all the cubes than in a complex cloud of points defining them. 😊

**Reply by MatthewFlamm** [reply]
The bounds are two points in Euclidean space.  The ordering of `(xmin, xmax, ymin, ymax, zmin, xmax)` is a vtk convention, so I think it would be unlikely to change here. You can mentally use `(xmin, ymin, zmin)` and `(xmax, ymax, xmax)`.  You can write your own conversion function to keep things in your preferred order.

**Reply by juanccilleruelo** [reply]
> The bounds are two points in Euclidean space. The ordering of `(xmin, xmax, ymin, ymax, zmin, xmax)` is a vtk convention, so I think it would be unlikely to change here. You can mentally use `(xmin, ymin, zmin)` and `(xmax, ymax, xmax)`. You can write your own conversion function to keep things in your preferred order.

What is desirable for me is a boolean parameter that says if the drawn object will be centered.

The current behavior can be considered "centered = true," and the solution I proposed "centered = false."

Why is this useful? Because I came from OpenSCAD and there, I have this option. No other reason.  😉

I'm grateful for your responses. Thanks.
